### PR TITLE
Drawing Feature  [ #6 ]( https://github.com/witseie-elen4010/2024-group-lab-003/issues/6 )

### PR DESCRIPTION
**Pull request by akivalevitt**
# Drawing Feature

## Description 

Created a basic drawing page where players, both admin and regular are able to draw the picture on the canvas provided.

The player can choose any pen color from a palette of all possible colors.

## Acceptance Criteria

1. The player is able to draw on the canvas with the cursor
2. The player is able to select any colors as the pen
3. The player is able to clear the canvas and start drawing from scratch
4. The player can save the image created as a PNG

## Story Points
**3** Story points

### CODE REVIEW

File: src/views/drawing.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet">
+</head>
+<body class="bg-light d-flex align-items-center" style="height: 100vh;">
+    <div class="container text-center">
+        <div class="toolbar mb-3">
+            <!-- <button id="black" class="btn btn-primary">Black</button>
+            <button id="red" class="btn btn-primary">Red</button>
+            <button id="green" class="btn btn-primary">Green</button>
+            <button id="blue" class="btn btn-primary">Blue</button> -->
+            <input type="color" id="colorPicker" value="#000000" class="form-control form-control-color">
```
Comment (Yishai-Bass) [inline src/views/drawing.html:16]: Brilliant work on this Akiva! I think though that the colour palette is a bit out of the scope of this user story and it should be defined as its own granular event. Consider defining a new user story for this.

File: src/views/drawing.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            <button id="clear" class="btn btn-primary">Clear</button>
```
Comment (Yishai-Bass) [inline src/views/drawing.html:17]: Again well done for this but it seems that this is also out of the scope of this user story.

File: src/views/drawing.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <button id="save" class="btn btn-primary">Save Drawing</button>
```
Comment (Yishai-Bass) [inline src/views/drawing.html:18]: Good job implementing the saving feature but consider making this its own user story.

**Review by Yishai-Bass — CHANGES_REQUESTED** [review]
Design: Approved
The implementation of the drawing feature on the canvas is well-integrated into the web application's interface. The setup for handling drawing states and interactions is efficient, though adding touch event support could enhance usability on touch-enabled devices.

Functionality: Denied. 
Out of scope of a single user story

Complexity: Approved
The code is straightforward and manages the complexity well. The drawing logic is encapsulated within its own function, which simplifies modifications and potential extensions like adding different colors or pen sizes.

Tests: Approved. 
No tests present but still early in the development process and is okay for now.

Naming: Approved
Naming conventions are clear and descriptive. Names like penColor, penWidth, and painting effectively describe their purposes and enhance code readability.

Comments: Approved

Style: Approved
The code adheres to common JavaScript conventions, such as using const for variables that are not re-assigned, and arrow functions for event handlers. The style is consistent, which helps in maintaining the code.

Documentation: Approved

File: src/views/drawing.html (hunk 1 above)
Comment (akivalevitt) [reply src/views/drawing.html]: issue fixed

File: src/views/drawing.html (hunk 2 above)
Comment (akivalevitt) [reply src/views/drawing.html]: issue fixed

File: src/views/drawing.html (hunk 3 above)
Comment (akivalevitt) [reply src/views/drawing.html]: issue fixed

**Review by Yishai-Bass — APPROVED** [review]
Design: Approved
Functionality: Approved
Complexity: Approved
Tests: Approved. No tests present but still early in the process
Naming: Approved
Comments: Approved
Style: Approved
Documentation: Approved

**Pull request merged**